File: frontend/src/api/handover.ts
import { api } from '../app/utils/api';

export interface HandoverRecord {
  id: number;
  title: string;
  outgoingUserId?: number;
  outgoingUserName?: string;
  incomingUserId?: number;
  incomingUserName?: string;
  status: string;
  statusLabel: string;
  assetCount: number;
  workorderCount: number;
  approvalCount: number;
  summary?: string;
  riskNote?: string;
  createdAt?: string;
  updatedAt?: string;
}

export interface HandoverList {
  records: HandoverRecord[];
  total: number;
}

export interface HandoverMeta {
  statuses: string[];
  readOnlyNotice: string;
}

export interface HandoverQuery {
  page?: number;
  pageSize?: number;
  status?: string;
  keyword?: string;
}

export function listHandoverTasks(params: HandoverQuery = {}) {
  return api.get<HandoverList>('/system/handover', { params });
}

export function getHandoverDetail(id: number) {
  return api.get<HandoverRecord>(`/system/handover/${id}`);
}

export function getHandoverMeta() {
  return api.get<HandoverMeta>('/system/handover/meta');
}

File: backend/src/main/java/com/ams/entity/Handover.java
package com.ams.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("handover")
public class Handover {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String tenantId;
    private String title;
    private Long outgoingUserId;
    private String outgoingUserName;
    private Long incomingUserId;
    private String incomingUserName;
    /** PENDING/IN_PROGRESS/COMPLETED/CANCELLED */
    private String status;
    private Integer assetCount;
    private Integer workorderCount;
    private Integer approvalCount;
    private String summary;
    private String riskNote;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}
